packages/@react-stately/table/src/TableUtils.ts

```diff
@@ -133,42 +65,260 @@ export interface IColumn {
+ *
+ * There are likely faster ways to do this, I've chosen to stick to the spec as closely as possible for readability, accuracy, and for the
+ * note that this behaving quadratically is unlikely to be a problem.
+ * @param availableWidth - The visible width of the table.
+ * @param columns - The table defined columns.
+ * @param changedColumns - Any columns we want to override, for example, during resizing.
+ * @param getDefaultWidth - A function that returns the default width of a column by its index.
+ * @param getDefaultMinWidth - A function that returns the default min width of a column by its index.
+ */
 export function calculateColumnSizes(availableWidth: number, columns: IColumn[], changedColumns: Map<Key, ColumnSize>, getDefaultWidth, getDefaultMinWidth) {
-  let remainingSpace = availableWidth;
-  let {staticColumns, dynamicColumns} = columns.reduce((acc, column, index) => {
+  let flexItems = columns.map((column, index) => {
     let width = changedColumns.get(column.key) != null ? changedColumns.get(column.key) : column.width ?? column.defaultWidth ?? getDefaultWidth?.(index) ?? '1fr';
-    let minWidth = column.minWidth ?? getDefaultMinWidth?.(index);
-    column.minWidth = minWidth;
-
+    let frozen = false;
+    let baseSize = 0;
+    let flex = 0;
     if (isStatic(width)) {
-      let w = parseStaticWidth(width, availableWidth);
-      w = Math.max(
-        getMinWidth(column.minWidth, availableWidth),
-        Math.min(Math.floor(w), getMaxWidth(column.maxWidth, availableWidth)));
-      acc.staticColumns.push({index, column, width: w} as IndexedColumn);
-      acc.dynamicColumns.push(null);
-      remainingSpace -= w;
+      baseSize = parseStaticWidth(width, availableWidth);
+      frozen = true;
     } else {
-      let w = parseFractionalUnit(width);
-      acc.staticColumns.push(null);
-      acc.dynamicColumns.push({index, column, width: w} as IndexedColumn);
+      flex = parseFractionalUnit(width);
+      if (flex <= 0) {
+        frozen = true;
+      }
     }
-    return acc;
-  }, {staticColumns: [] as IndexedColumn[], dynamicColumns: [] as IndexedColumn[]});
-  let newColWidths = getDynamicColumnWidths(dynamicColumns, remainingSpace, availableWidth);
 
-  return staticColumns.map((col, i) => {
-    if (col) {
-      return col.width;
+    let min = getMinWidth(column.minWidth ?? getDefaultMinWidth?.(index) ?? 0, availableWidth);
+    let max = getMaxWidth(column.maxWidth, availableWidth);
+    let hypotheticalMainSize = Math.max(min, Math.min(baseSize, max));
+    return {
+      frozen,
+      baseSize,
+      hypotheticalMainSize,
+      min,
+      max,
+      flex,
+      targetMainSize: null,
+      scaledFlexShrinkFactor: 0,
+      violation: ''
+    };
+  });
+  // 9.7.1
+  let totalHypothetical = flexItems.reduce((acc, col) => acc + col.hypotheticalMainSize, 0);
+  let isGrow = false;
+  if (totalHypothetical < availableWidth) {
+    isGrow = true;
```

Comment:
Commenting out the if statement and hard coding `isGrow` to true still passes all of the tests. Do we need any of the code for the `!isGrow` case or are tests missing?

Reply:
This is probably one of those places where we can simplify. Flex shrink would kick in if a base size was bigger than zero. For instance `flex: 1 1 auto` or `flex: 1 1 500px`, but since we do not care about the content size (auto) or any other starting size other than 0, only grow will apply. We always do `flex: 1 1 0` or `flex: n n 0`, either way the basis is always 0.

From this perspective, we could probably simplify further since `flex: 1 0 0` should be equivalent to `flex: 1 1 0`.